gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -347,6 +343,30 @@ public AddSpecResponse onAddSpec(Spec addedSpec) {
     return new AddSpecResponse<>(response);
   }
 
+  /**
+   * Remove a flowSpec from schedule due to another leader being elected
+   * Unlike onDeleteSpec, we want to avoid deleting the flowSpec on the executor
+   * and we still want to unschedule if we cannot connect to zookeeper as the current node cannot be the master
+   * @param specURI
+   * @param specVersion
+   */
+  private void unscheduleSpec(URI specURI, String specVersion) {
```

Comment:
Could we do this instead by just adding an overloaded onDeleteSpec with a boolean option to remove from the orchestrator to avoid the code duplication?

Reply:
or call this from inside onDeleteSpec()

Reply:
onDeleteSpec also checks for connection to zookeeper though, and I think that's a bad check if the host is demoted. If a host that used to be master gets demoted, it's likely struggling with connectivity so unscheduleSpec handles that.

@arjun4084346 can do that too